File: services/arrecadacao-api/arrecadacao-application/src/main/java/br/com/ecad/arrecadacao/application/audit/NoopAuditContextProvider.java
package br.com.ecad.arrecadacao.application.audit;

import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
import org.springframework.stereotype.Component;

@Component
@ConditionalOnMissingBean(AuditContextProvider.class)
public class NoopAuditContextProvider implements AuditContextProvider {
    @Override
    public AuditContext current(String fallbackUsername) {
        return AuditContext.system(fallbackUsername);
    }
}

File: services/cadastro-api/2-Application/Cadastro.Application/Audit/NoopAuditContextProvider.cs
using Ecad.Audit.Sdk;

namespace Cadastro.Application.Audit;

public sealed class NoopAuditContextProvider : IAuditContextProvider
{
    public AuditContext Current(string fallbackUsername = "system")
    {
        var requestId = Guid.NewGuid().ToString();
        return new AuditContext(
            UserId: fallbackUsername,
            Username: fallbackUsername,
            DisplayName: fallbackUsername,
            Roles: [],
            AuthProvider: "system",
            TraceId: requestId.Replace("-", string.Empty, StringComparison.Ordinal),
            RequestId: requestId,
            Channel: "SYSTEM");
    }
}
